src/ctap2/mod.rs

```
@@ -809,5 +809,440 @@ pub(crate) fn set_or_change_pin_helper(
+                        );
+
+                        if remaining_samples == 0 {
+                            if let BioEnrollmentCmd::StartNewEnrollment(Some(ref name)) = command {
+                                bio_cmd.subcommand = BioEnrollmentCommand::SetFriendlyName((
+                                    template_id.into_vec(),
+                                    name.clone(),
+                                ));
+                                unwrap_result!(bio_cmd.regenerate_puap(), callback);
+                                continue;
+                            } else {
+                                callback.call(Ok(ManageResult::Success));
+                                return true;
+                            }
+                        } else {
+                            bio_cmd.subcommand = BioEnrollmentCommand::EnrollCaptureNextSample((
+                                template_id,
+                                timeout,
+                            ));
+                            unwrap_result!(bio_cmd.regenerate_puap(), callback);
+                            continue;
+                        }
+                    }
+                    BioEnrollmentCommand::EnumerateEnrollments => {
+                        let list = result.template_infos.iter().map(|x| x.into()).collect();
+                        callback.call(Ok(ManageResult::BioEnrollment(
+                            BioEnrollmentResult::EnrollmentList(list),
+                        )));
+                        return true;
+                    }
+                    BioEnrollmentCommand::SetFriendlyName(_)
+                    | BioEnrollmentCommand::RemoveEnrollment(_)
+                    | BioEnrollmentCommand::CancelCurrentEnrollment => {
+                        callback.call(Ok(ManageResult::Success));
+                        return true;
+                    }
+                    BioEnrollmentCommand::GetFingerprintSensorInfo => todo!(),
```

`todo!` will panic so let's return an error here instead.

---

Or better yet: let me actually implement this :grimacing: